src/utilities/paramsFinder.ts

```diff
@@ -30,10 +31,27 @@ export function setParamValuesOnUrl(path: string, paramReplace?: ParamReplace):
 }
 
 function getParamRegexPattern(path: string, paramName: string): RegExp {
+  const regexPattern = [
+    replaceOptionalParamSyntaxWithCaptureGroup,
+    replaceREquiredParamSyntaxWithCaptureGroup,
+    replaceParamSyntaxWithCatchAlls,
```

Comment:
I moved the logic for individual replace logic into their own functions then added the `replaceParamSyntaxWithCatchAlls` function from routeRegex.